src/dotnet/CarbonAware.Tools.WattTimeClient/Configuration/WattTimeClientConfiguration.cs

```diff
@@ -0,0 +1,36 @@
+
+namespace CarbonAware.Tools.WattTimeClient.Configuration;
+
+/// <summary>
+/// A configuration class for holding WattTime client config values.
+/// </summary>
+public class WattTimeClientConfiguration
+{
+    public const string Key = "WattTimeClient";
+
+    /// <summary>
+    /// Gets or sets the username to use when connecting to WattTime.
+    /// </summary>
+    public string? Username { get; set; }
+
+    /// <summary>
+    /// Gets or sets the password to use when connecting to WattTime
+    /// </summary>
+    public string? Password { get; set; }
+
+    /// <summary>
+    /// Validate that this object is properly configured.
+    /// </summary>
+    public void Validate()
+    {
+        if (string.IsNullOrEmpty(this.Username))
```

Comment:
>IsNullOrEmpty [](http://example.com/codeflow?start=19&length=13)

I tend to prefer `IsNullOrWhiteSpace` nowadays, unless blank-chars are truly are a valid input